### ISSUE TITLE
About Separating Instrumental

### ISSUE THREAD

**Issue by asimptot** [ReQEnoxus/vocal-separator#1]
Hello,

I would like to separate instrumental wav, as well. How can I do that using your repository?

**Comment by ReQEnoxus** [comment]
Hello and sorry for late response. I guess, the easiest way would be to invert the binary mask produced by the model. That way, it will zero out everything it considered to be vocal, so the rest would be instrumentals. To do that, simply change 52th line in [analyzer.py](https://github.com/ReQEnoxus/vocal-separator/blob/master/analyzer/analyzer.py):
From:
```python
binary = np.delete(binary, np.s_[song.stft.shape[1]:binary.shape[1]], 1)
```
To:
```python
binary = 1 - np.delete(binary, np.s_[song.stft.shape[1]:binary.shape[1]], 1)
```

**Comment by asimptot** [comment]
Hi,

Thanks, when i converted in the following raw, it extracted to
instrumental.

            elif vocal_prediction > vocal_sensitivity: -->
            elif vocal_prediction < vocal_sensitivity:

Btw, how do you decide tf.keras.models values? Is there any reference?

            tf.keras.layers.Conv2D(32, (3, 3), padding='same',
input_shape=(1025, 25, 1)),
             tf.keras.layers.LeakyReLU(),
             tf.keras.layers.Conv2D(16, (3, 3), padding='same'),
             tf.keras.layers.LeakyReLU(),
             tf.keras.layers.MaxPooling2D(pool_size=(3, 3)),
             tf.keras.layers.Dropout(0.25),

             tf.keras.layers.Conv2D(64, (3, 3), padding='same'),
             tf.keras.layers.LeakyReLU(),
             tf.keras.layers.Conv2D(16, (3, 3), padding='same'),
             tf.keras.layers.LeakyReLU(),
             tf.keras.layers.MaxPooling2D((3, 3)),
             tf.keras.layers.Dropout(0.25),

             tf.keras.layers.Flatten(),
             tf.keras.layers.Dense(128),
             tf.keras.layers.LeakyReLU(),
             tf.keras.layers.Dropout(0.5),
             tf.keras.layers.Dense(1025, activation='sigmoid')



31 Ağu 2022 Çar 18:48 tarihinde ReQEnoxus ***@***.***> şunu
yazdı:

**Comment by ReQEnoxus** [comment]
Yes, this project is heavily inspired by [this article](https://towardsdatascience.com/audio-ai-isolating-vocals-from-stereo-music-using-convolutional-neural-networks-210532383785). In fact, my model is pretty much the same but slightly adjusted to fit my dataset

**Comment by asimptot** [comment]
Which article? If you share with me, i'll be glad! :)

31 Ağu 2022 Çar 19:21 tarihinde ReQEnoxus ***@***.***> şunu
yazdı:

**Comment by ReQEnoxus** [comment]
The link should've been clickable) Here's the raw link: https://towardsdatascience.com/audio-ai-isolating-vocals-from-stereo-music-using-convolutional-neural-networks-210532383785

**Comment by asimptot** [comment]
Thanks, i will look into the article. You used pre trained model to import
weights, right? If so, the pretrained model was created by musdb18 db file?
I saw some code raws have been related to musdb18. Is it possible to
optimize output quality? When i checked the vocal songs, the sound was
interrupted many times.

31 Ağu 2022 Çar 19:24 tarihinde ReQEnoxus ***@***.***> şunu
yazdı:

**Comment by ReQEnoxus** [comment]
The model was trained by myself using the full version of musdb18 dataset. Unfortunately, I've lost the code that generated training inputs so I couldn't include the training part in repo (so I've just included pre-trained weights to make it working). In order to improve the quality, you could probably use this model and weights as starting point and train it with some other dataset following ideas in the article. You could also play around with arguments to fit better for a particular song

**Comment by asimptot** [comment]
Thanks, it will be very helpful!

31 Ağu 2022 Çar 19:46 tarihinde ReQEnoxus ***@***.***> şunu
yazdı:
